Encountify.Android/CustomMapRenderer.cs

```diff
@@ -43,116 +56,145 @@ protected override void OnElementChanged(ElementChangedEventArgs<Map> e)
-                var infoDistance = view.FindViewById<TextView>(Resource.Id.InfoWindowDistance);
-
-                if (infoPoints != null)
-                {
-                    infoPoints.Text = "100 POINTS✨";
-                }
-                if (infoTitle != null)
-                {
-                    infoTitle.Text = marker.Title;
-                }
-                if (infoSubtitle != null)
+                if (double.TryParse(distanceStringList[0], out var distanceDouble))
                 {
-                    infoSubtitle.Text = marker.Snippet;
-                }
-                if (infoDistance != null)
-                {
-                    infoDistance.Text = distance + " away";
-                }
+                    if (distanceDouble <= 30 && distanceStringList[1] == "m")
+                    {
+                        var layout = Resource.Layout.VisitedInfoWindow; //TODO make the VisitedInfoWindow actually look good
 
-                return view;
+                        return GetInfoWindowView(layout, inflater, marker, message: "You're here. PRESS ME!");
+                    }
+                    else //TODO make a InfoWindow for the markers that have already been visited my the user 
+                    {
+                        var layout = Resource.Layout.MapInfoWindow;
+
+                        return GetInfoWindowView(layout, inflater, marker, distance: distanceString);
+                    }
+                }
             }
 
             return null;
         }
 
-        async void OnInfoWindowClick(object sender, GoogleMap.InfoWindowClickEventArgs e)
+        public View GetInfoWindow(Marker marker)
         {
-            var access = new DatabaseAccess<Location>();
-            var locationList = access.GetAllAsync().Result;
+            View view = GetInfoWindowAsync(marker).Result;
 
-            foreach (var s in locationList)
-            {
-                if (s.Name == e.Marker.Title)
-                {
-                    var id = s.Id;
-                    await Shell.Current.GoToAsync($"{nameof(LocationDetailPage)}?{nameof(LocationDetailViewModel.Id)}={id}");
-                    break;
-                }
-            }
+            return view;
         }
 
-        void OnInfoWindowClose(object sender, GoogleMap.InfoWindowCloseEventArgs e)
+        async void OnInfoWindowClick(object sender, GoogleMap.InfoWindowClickEventArgs e)
         {
-            CurrentPinWindow = null;
-        }
+            Locations pinLocation = new Locations(e.Marker.Position.Latitude, e.Marker.Position.Longitude);
+            var distanceString = await DistanceCounter.GetFormattedDistance(pinLocation);
 
-        void RefreshWindow(CustomPin pin)
-        {
-            Device.BeginInvokeOnMainThread(delegate ()
+            var distance = distanceString.Split(" ");
+
+            if (double.TryParse(distance[0], out var distanceDouble))
```

Comment:
I think it would be a good idea to add additional function to DistanceCounter, that only compares distance, or returns floating point value. Currently it does not work when you set distance to yards in the settings, though now is probably not a good time, since we need to complete requirements.

Reply:
Yes I think it would be a good idea eventually